src/main/kotlin/com/ampnet/userservice/persistence/StringCryptoConverter.kt

```diff
@@ -0,0 +1,77 @@
+package com.ampnet.userservice.persistence
+
+import com.ampnet.userservice.exception.EncryptionException
+import com.ampnet.userservice.exception.ErrorCode
+import org.springframework.beans.factory.annotation.Value
+import java.security.GeneralSecurityException
+import java.security.InvalidAlgorithmParameterException
+import java.security.InvalidKeyException
+import java.security.NoSuchAlgorithmException
+import java.util.Base64
+import javax.crypto.BadPaddingException
+import javax.crypto.Cipher
+import javax.crypto.IllegalBlockSizeException
+import javax.crypto.NoSuchPaddingException
+import javax.persistence.AttributeConverter
+import javax.persistence.Converter
+
+@Converter
+class StringCryptoConverter : AttributeConverter<String, String> {
+
+    private val cipherInitializer = CipherInitializer()
+
+    @Value("\${spring.flyway.placeholders.private_key}")
+    private lateinit var databaseEncryptionKey: String
+
+    @Throws(
+        NoSuchAlgorithmException::class,
+        InvalidKeyException::class,
+        InvalidAlgorithmParameterException::class,
+        BadPaddingException::class,
+        NoSuchPaddingException::class,
+        IllegalBlockSizeException::class
+    )
```

Comment:
Looks to me it only throws `EncryptionException`.

Reply:
prepareAndInitCipher function on CipherInitializer throws those other exceptions. I can leave the declaration only on the function throwing it.